backends/platform/sdl/macosx/macosx.cpp

```diff
@@ -117,6 +117,20 @@ void OSystem_MacOSX::initBackend() {
 	_textToSpeechManager = new MacOSXTextToSpeechManager();
 #endif
 
+	// Migrate savepath.
+	// It used to be in ~/Documents/ScummVM Savegames/, but was changed to use the application support
+	// directory. To migrate old config files we use a flag to indicate if the config file was migrated.
+	// This allows detecting old config files. If the flag is not set we:
+	// 1. Set the flag
+	// 2. If the config file has no custom savepath and has some games, we set the savepath to the old default.
```

Comment:
Perhaps we can check the successful migration from the value of the actual save path (e.g. if it starts with "~/Documents"), instead of setting a flag?

Reply:
The purpose of the flag is to make the distinction between a config file created with an older version of ScummVM with default savegame path and loaded for the first time in the new version, and a brand new config file created with the new version of ScummVM (and with default savegame path). In both cases the "savepath" is not specified in the config file, so there is no actial save path to check. But in one case we want to use the old default, and in the other the new default.

Other ideas I had was:
- Check the version saved in the config file. But `scummvm_main` update the version in the config file before the backends gets an opportunity to check it. Also even if I could get the version string before it is updated, it is not trivial to parse and determine if this is a version before or after the change.
- Check if the `~/Documents/ScummVM Savegames` folder exists. But that would mean accessing the Documents folder, which defeats the purpose of this change.

Reply:
Thanks for the clarification :)